rascal2/ui/model.py

```diff
@@ -103,9 +104,14 @@ def load_project(self, load_path: str):
                 "It may contain invalid parameter values or be invalid JSON."
             ) from err
 
-        # TODO add saving `Project` once this is possible
-        # https://github.com/RascalSoftware/python-RAT/issues/76
-        self.project = RAT.Project()
+        project_file = Path(load_path, "project.json")
+        try:
```

Comment:
I think this try excepts should be in the presenter instead of the model so that we can log the actual exception then notify the user

Reply:
at that point, the entire function may as well be in the presenter (as otherwise all we're doing on the model side is assigning some attributes), if that's what you think would work better? if you look at `dialogs/project_dialog.py`, you can see that the `LoadDialog.load_project` catches ValueErrors raised by `presenter.load_project` and notifies the user in the dialog

Reply:
ah i see. where should they be logged? this loading will (usually) be done in the startup dialog before a project has been loaded, so the 'project log' is not yet defined.

just to make sure we're on the same page, this is what i mean by notifies in the dialog: the same error you get if the folder doesn't contain a project, or the new project dialog finds a project already exists, etc. (none of which are logged)
![Screenshot 2024-11-08 11:32:48](https://github.com/user-attachments/assets/d02df55d-db8b-42c3-9259-22dc5b7b6c1e)

Reply:
I setup a crash.log (might need to be renamed) for when the project is not created so that could be used to log the error with some refactoring https://github.com/RascalSoftware/RasCAL-2/blob/f9b5964b80b4adb57cdb78e304955ad11dfaddca/rascal2/config.py#L99

Reply:
done: i also made `log_uncaught_exceptions` print to the terminal as well as other handlers when there's a crash, so i don't have to go digging through the log file to find the uncaught error!